jupyterhub/roles.py

```diff
@@ -192,57 +284,138 @@ def add_and_remove(db, obj, kind, current_role, new_role):
         add_and_remove(db, obj, kind, admin_role, user_role)
 
 
```

Comment:
Which methods in `roles.py` can be private?

Reply:
Quite a few actually, thanks! Updated.